Monika After Story/game/script-songs.rpy

```diff
@@ -0,0 +1,302 @@
+default persistent._mas_songs_database = dict()
+
+init -10 python in mas_songs:
+    # Event database for songs
+    song_db = {}
+
+    #Song type constants
+    TYPE_LONG = "long"
+    TYPE_SHORT = "short"
+
+init python in mas_songs:
+    import store
+    def checkRandSongDelegate():
+        """
+        Checks if we have random songs available, and if so unlocks the random delegate if locked
+        """
+        #Get ev
+        rand_delegate_ev = store.mas_getEV("monika_sing_song_random")
+
+        if rand_delegate_ev:
+            #Don't bother doing the rest if it's random already
+            if rand_delegate_ev.random:
+                return
+
+            #If we've got at least one random, then unlock the rand delegate
+            if hasRandomSongs():
+                rand_delegate_ev.random = True
+
+    def getUnlockedSongs():
+        """
+        Gets a list of all unlocked songs
+
+        OUT: list of all unlocked songs
```

Comment:
can you comment that the returned list is in a tuple format ready for a scrollable menu